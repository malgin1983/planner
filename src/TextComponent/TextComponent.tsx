import React from 'react';
import { View, Text, TouchableHighlight, SafeAreaView, StyleSheet } from 'react-native';
import { useSelector } from 'react-redux';
import { RootState, useAppDispatch } from '../redux/Store';
import { increment, decrement } from '../redux/CountReducer/Reducer';
import { fetchUserById } from '../redux/Thunks/CountActions';

interface ITextComponent {
    text: string;
}

const TextComponent: React.FC<ITextComponent> = (props) => {
    const dispatch = useAppDispatch();
    dispatch(fetchUserById(1));
    const count = useSelector((state: RootState) => state.counter.count);
    // const fetchData = useSelector((state: RootState) => state.counter.data);
    // console.log('Data from fetch', fetchData);
    const pressPluse = () => {
        dispatch(increment());
    };
    const pressMinus = () => {
        dispatch(decrement());
    };

    return (
        <SafeAreaView style={styles.container}>
            <View style={styles.container}>
                <View style={styles.textContainer}>
                    <Text testID={'text-component -props'}>{props.text}</Text>
                </View>
                <View style={styles.textContainer}>
                    <Text testID={'text-component'}>{`Count - ${count}`}</Text>
                </View>
                <View style={styles.btnBlock}>
                    <TouchableHighlight testID={'text-btn-minus'} style={styles.btn} onPress={pressMinus}>
                        <Text>{'minus'}</Text>
                    </TouchableHighlight>
                    <TouchableHighlight testID={'text-btn-plus'} style={styles.btn} onPress={pressPluse}>
                        <Text>{'plus'}</Text>
                    </TouchableHighlight>
                </View>
            </View>
        </SafeAreaView>
    );
};

export default TextComponent;

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    main: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'grey',
    },
    btn: {
        width: 100,
        height: 50,
        backgroundColor: 'green',
        alignItems: 'center',
        justifyContent: 'center',
        color: 'white',
    },
    text: {
        color: 'black',
        fontSize: 18,
    },
    textContainer: {
        marginBottom: 8,
        alignItems: 'center',
        justifyContent: 'center',
    },
    btnBlock: {
        flex: 1,
        width: 250,
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
});
