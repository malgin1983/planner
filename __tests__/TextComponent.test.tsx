import 'react-native';
import React from 'react';
import {create, act} from 'react-test-renderer';
import TextComponent from '../src/TextComponent/TextComponent';
const element = create(<TextComponent text={'Hello'} />);

describe('Test  - TextComponent', () => {
  it('Render TextComponent and Snapshot', () => {
    create(<TextComponent text={'Hello'} />);
    expect(element).toMatchSnapshot();
  });
  it('Press on  button', () => {
    const btn = element.root.findByProps({testID: 'text-btn'}).props;
    act(() => {
      btn.onPress();
    });
    const text = element.root.findByProps({testID: 'text-component'}).props;
    expect(text.children).toEqual('Touch 1 times');
  });
});
