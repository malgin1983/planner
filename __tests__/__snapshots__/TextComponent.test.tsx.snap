// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test  - TextComponent Render TextComponent and Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 8,
      }
    }
  >
    <Text
      testID="text-component"
    >
      Touch 0 times
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "green",
        "color": "white",
        "height": 50,
        "justifyContent": "center",
        "width": 100,
      }
    }
    testID="text-btn"
  >
    <Text>
      Hello
    </Text>
  </View>
</View>
`;
